chore(core): removed `system, pagesetup` event

jdalsem opened the pull request:
All previously registered events have been moved to new locations..
mostly menu register hooks.

BREAKING CHANGE:
You can no longer use the `system, pagesetup` event

fixes #9733
- [x] document the removed functions
- [x] check usage of removed functions

jdalsem committed:
chore(core): removed `system, pagesetup` event

All previously registered events have been moved to new locations..
mostly menu register hooks.

BREAKING CHANGE:
You can no longer use the `system, pagesetup` event

mrclay commented on engine/lib/admin.php:
```diff
@@ -196,7 +196,11 @@ function _elgg_create_notice_of_pending_upgrade($event, $type, $object) {
+	$url = elgg_get_simplecache_url('admin.css');
+	elgg_register_css('elgg.admin', $url);
+		
+	elgg_register_plugin_hook_handler('register', 'menu:admin_header', '_elgg_admin_header_menu');
```
Just thinking out loud: I wonder if we should register all these (formally using elgg_register_menu_item) with a low priority, like 1. Consider the case where a 2.x plugin is also registering for this hook, and expects the menu items already to be present. It's more of an issue in the plugins, because their inits potentially get called after a 3rd party plugin.

jdalsem replied:
> registering for this hook, and expects the menu items already to be present

it's an edge case, but could easily be fixed...

mrclay replied:
True. A mention in the upgrade docs would be sufficient. E.g. "Many core menu items are now added to the menu later, in the `register` hook. If your plugin looks for/alters core menu items, you may need to use the `register` hook with a higher priority."

jdalsem replied:
i was typing when you typed the samed

mrclay commented on mod/groups/start.php:
```diff
@@ -123,6 +122,9 @@ function groups_init() {
 	elgg_register_plugin_hook_handler('register', 'menu:filter:groups/all', 'groups_setup_filter_tabs');
+
+	elgg_register_plugin_hook_handler('register', 'menu:page', '_groups_page_menu_group_profile');
+	elgg_register_plugin_hook_handler('register', 'menu:page', '_groups_page_menu');
```
Looks like these handlers could be combined. Any reason for 2 functions?

mrclay commented:
Pretty exciting!

mrclay reviewed (changes requested):
The gear menu in developers fails to load.

jdalsem commented:
> The gear menu in developers fails to load.

huh, how would that be related to this PR?

mrclay commented on mod/developers/start.php:
```diff
@@ -81,33 +82,70 @@ function developers_process_settings() {
 	}
 }
 
-function developers_setup_menu() {
```
This is still used in mod/developers/views/default/developers/gear_popup.php

mrclay replied:
Try just removing the call in that view.

mrclay replied:
May be wise to search for all functions you're removing here.

jdalsem replied:
lol... you liked your own comment :)

i added a check for all removed functions as a todo

jdalsem commented:
what is our/your stance on the use of the private functions in the plugins? with the underscores? is that the way to go?

mrclay commented:
When I'm not lazy I prefix with underscores, but... I don't know if it really matters. We should just document that all plugin functions are private unless specifically marked `@api`.

mrclay commented:
⬆️  I made an issue on this.

jdalsem commented:
Updated the PR
- fixed developers gear issue
- documented removed function (if not marked private)
- checked usage of removed functions

Ready to be merged

jeabakker commented on docs/appendix/faqs/development.rst:
```diff
@@ -149,10 +149,11 @@ There are 5 :doc:`Elgg events </design/events>` that are triggered on every page
 
-The *boot*, *system* event is triggered before the plugins get loaded. There does not appear to be any difference between the timing of the next two events: *plugins_boot*, *system* and *init*, *system* so plugins tend to use *init*, *system*. This event is triggered in ``Elgg\Application::bootCore``. The *pagesetup*, *system* event is thrown the first time ``elgg_view()`` is called. Some pages l ...
+
+The *boot*, *system* event is triggered before the plugins get loaded. There does not appear to be any difference between the timing of the next two events: *plugins_boot*, *system* and *init*, *system* so plugins tend to use *init*, *system*. This event is triggered in ``Elgg\Application::bootCore``. The *shutdown*, *system* event is triggered after the page has been sent to the requester and is ...
```
You forgot to document the use of the ``ready`` ``system`` event

jdalsem replied:
added the ready event docs

jeabakker commented on docs/guides/events-list.rst:
```diff
@@ -21,10 +21,6 @@ System events
-    Called just before the first content is produced. Is triggered by ``elgg_view()``.
-    Use the menu or page shell hooks instead.
-
 **shutdown, system**
```
also add ``ready`` ``system`` explanation

jdalsem replied:
was already available 2 lines above...
